File: frontmv2/src/app/model/hospital.ts
import { HospitalDTO } from './dto/hospitalDTO';
import { Endereco } from './endereco';
import { Profissional } from './profissional';

export class Hospital {
  id: number
  nome: string
  telefone: string

  endereco: Endereco

  profissionais: Profissional[]

  static withId(id: number): Hospital {
    const hospital = new Hospital();
    hospital.id = id
    return hospital
  }
  static fromDTO(dto: HospitalDTO): Hospital {
    let hospital = new Hospital()
    hospital.id = dto.id
    hospital.nome = dto.nome
    hospital.telefone = dto.telefone
    hospital.profissionais = dto.profissionais
    let endereco = new Endereco();
    endereco.rua = dto.rua
    endereco.numero = dto.numero
    endereco.bairro = dto.bairro
    endereco.cidade = dto.cidade
    endereco.estado = dto.estado
    hospital.endereco = endereco

    return hospital
  }

  toDTO(): HospitalDTO {
    let dto = new HospitalDTO()
    dto.id = this.id
    dto.nome = this.nome
    dto.telefone = this.telefone
    dto.profissionais = this.profissionais
    dto.rua = this.endereco.rua
    dto.numero = this.endereco.numero
    dto.bairro = this.endereco.bairro
    dto.cidade = this.endereco.cidade
    dto.estado = this.endereco.estado



    return dto
  }

}

File: mvbackteste/src/main/java/br/com/testeMV/model/Hospital.java
package br.com.testeMV.model;

import br.com.testeMV.model.dto.HospitalDTO;
import lombok.*;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table
@Entity
public class Hospital {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "hospital_id")
	private Long id;

	private String nome;
	private String telefone;

	@Embedded
	private Endereco endereco;

	@ManyToMany(fetch = FetchType.EAGER)
	@With
	private Set<Profissional> profissionais = new HashSet<>();

	public HospitalDTO toDTO() {
		return HospitalDTO.builder()
				.id(id)
				.nome(nome)
				.telefone(telefone)
				.profissionais(profissionais)
				.rua(endereco.getRua())
				.numero(endereco.getNumero())
				.bairro(endereco.getBairro())
				.cidade(endereco.getCidade())
				.estado(endereco.getEstado())
				.build();
	}


}
